main.go

```diff
@@ -18,36 +18,65 @@ limitations under the License.
 
 import (
 	"context"
+	"fmt"
 	"log"
 	"os"
 
-	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/aws/credentials"
-	"github.com/aws/aws-sdk-go/aws/session"
-	awsSqs "github.com/aws/aws-sdk-go/service/sqs"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	awsSqs "github.com/aws/aws-sdk-go-v2/service/sqs"
+
 	"github.com/numaproj/numaflow-go/pkg/sourcer"
 
 	"github.com/numaproj-contrib/aws-sqs-source-go/pkg/sqs"
 )
 
-func initSess() *session.Session {
-	sess := session.Must(session.NewSessionWithOptions(session.Options{
-		Config: aws.Config{
-			Region:      aws.String(os.Getenv("AWS_REGION")),
-			Credentials: credentials.NewStaticCredentials(os.Getenv("AWS_ACCESS_KEY"), os.Getenv("AWS_SECRET"), ""),
-			Endpoint:    aws.String(os.Getenv("AWS_END_POINT")),
-		},
-		SharedConfigState: session.SharedConfigDisable,
-	}))
-	return sess
+const (
+	awsEndpointURL = "AWS_ENDPOINT_URL"
+	sqsQueueName   = "AWS_SQS_QUEUE_NAME"
+)
+
+func initClient(ctx context.Context, awsEndpoint string) (*awsSqs.Client, error) {
+	// Load default configs for aws based on env variable provided based on
+	// https://aws.github.io/aws-sdk-go-v2/docs/configuring-sdk/#specifying-credentials
+	cfg, err := config.LoadDefaultConfig(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed loading aws config, err: %v", err)
+	}
+	var client *awsSqs.Client
+	if awsEndpoint != "" {
+		client = awsSqs.NewFromConfig(cfg, func(options *awsSqs.Options) {
+			options.BaseEndpoint = aws.String(awsEndpoint)
+		})
+	} else {
+		client = awsSqs.NewFromConfig(cfg)
+	}
+	return client, nil
+}
+
+func getQueueUrl(client *awsSqs.Client, sqsQueueName string, ctx context.Context) (*awsSqs.GetQueueUrlOutput, error) {
+	// generate the queue url to publish data to queue via queue name.
+	queueURL, err := client.GetQueueUrl(ctx, &awsSqs.GetQueueUrlInput{QueueName: aws.String(sqsQueueName)})
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate SQS Queue url, err: %v", err)
+	}
+	return queueURL, nil
 }
 
 func main() {
-	sqsClient := awsSqs.New(initSess())
-	awsSqsSource, err := sqs.NewAWSSqsSource(sqsClient, os.Getenv("AWS_QUEUE"))
+	endPoint := os.Getenv(awsEndpointURL)
+	queueName := os.Getenv(sqsQueueName)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	client, err := initClient(ctx, endPoint)
+	if err != nil {
+		log.Fatalf("error in creating sqs client %s", err)
+	}
+	queue, err := getQueueUrl(client, queueName, ctx)
```

Comment:
```suggestion
	queue, err := client.GetQueueUrl(ctx, &awsSqs.GetQueueUrlInput{QueueName: aws.String(queueName)})
```

`getQueueUrl` is a [shallow](https://medium.com/@alexturdean/shallow-functions-the-silent-killer-of-good-software-527a29ab56c9) function. Please remove it.